lib/segment/src/id_tracker/point_mappings.rs

```diff
@@ -103,6 +103,11 @@ impl PointMappings {
             PointIdType::Uuid(uuid) => self.external_to_internal_uuid.remove(&uuid),
         };
 
+        // Also reset inverse mapping
+        if let Some(internal_id) = internal_id {
+            self.internal_to_external[internal_id as usize] = PointIdType::NumId(u64::MAX);
+        }
```

Comment:
Also resetting the inverse mapping is not strictly necessary. However, I think it's nice to do because that'll put the mapping in exactly the same state as before inserting the point.

Adding this here makes the new test `test_point_deletion_persists_reload()` happy. The alternative would be to not reset the inverse mapping and relax the test a bit to not assert raw values.